content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Description: provide the dynamic list content from available credentials
+# for Ansible Provider
+#
+module ManageIQ
+  module Automate
+    module AutomationManagement
+      module AnsibleTower
+        module Operations
+          class AvailableCredentials
+            AUTH_CLASS = "ManageIQ_Providers_AutomationManager_Authentication".freeze
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def main
+              fill_dialog_field(fetch_list_data)
+            end
+
+            private
+
+            def provider
+              # for provisioning we use     service_template
+              # for reconfig and retire use service
+              service = @handle.root['service_template'] || @handle.root['service']
+              service.try(:job_template, 'Provision').try(:manager)
```

Comment:
@gmcculloug I discussed this with @bzwei and this is where we came up with the the restriction that all job templates in a service have to point to the same manager. Since credentials are connected to a manager for the entire job template we will have a single manager. So all service_actions (Provision, Reconfigure, Retire) will point to the same manager and hence the same set of credentials.